src/core_plugins/kibana/public/dashboard/components/panel/lib/visualization.js

```diff
@@ -11,6 +11,22 @@ export default function visualizationLoader(savedVisualizations, Private) { // I
       // $scope.state comes via $scope inheritence from the dashboard app. Don't love this.
       savedVis.vis.listeners.click = filterBarClickHandler($scope.state);
       savedVis.vis.listeners.brush = brushEvent($scope.state);
```

Comment:
wrt. this snippet and the one in editor.js. 

Let's not dynamically add new API to the type. It distributes internal API over multiple files, and makes navigation in  the code harder. So, since  `getFilters` and `getQuery` are part of Vis, they should be defined on `Vis.prototype` (ui/public/vis/vis). 

I understand that we are trapping state in the closures here. If we need to inject this, we can add another method, something like `setFilters` or `setQuery` or `registerFilter`, and then call those from here.